aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,125 @@
-.. include:: /../common/stub-topic.txt
+Model Validation 
+================
+By `Rachel Appel <http://github.com/rachelappel>`_
 
-|stub-icon| Model Validation
-============================
+In this article:
 
-.. include:: /../common/stub-notice.txt
+- `Introduction to model validation`_
+- `Data annotations`_
+- `Model state`_
+- `Custom validation`_
+- `Client side validation`_ 
+- `Remote valiation`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/116
+Introduction to model validation
+--------------------------------
+Before an app stores data in a database, the app must validate the data. Data must be scrubbed for security, verified that it is appropriately formatted by type and size, and it must conform to your rules. Validation is necessary although it can be redundant and tedious to implement. In MVC, validation happens on both the client and server. 
+
+By using data annotations in MVC apps you can validate data by customizing models so they mimic the underlying structure of the data store. This way you can enforce integrity rules and constraints against models by applying attributes to model properties. These attributes generate validation code, thereby reducing the amount of code you must write. Fortunately, .NET has abstracted validation into the data annotations, often reducing validation to a single line of code.
+
+Data annotations
+----------------
+Data annotations are a way to configure model validation so it's similar to validation on fields in tables. This includes constraints such as assigning data types, required fields, or a primary or foreign key. Other types of validation include applying patterns to data to enforce integrity, such as a credit card, phone number, or email address (regular expressions). Data annotations do all this for you automatically.
```

Comment:
Avoid mixing a broader discussion of the [`System.ComponentModel.DataAnnotations` namespace](https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396) into this document. Server-side validation involves _only_ [`IValidatableObject`](https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.ivalidatableobject(v=vs.110).aspx) implementations, [`ValidationAttribute`](https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.validationattribute(v=vs.110).aspx) and its subclasses. In particular "primary or foreign keys" are not relevant, nor is "enforcing integrity".